website/source/docs/internals/scheduling.html.md

```diff
@@ -9,88 +9,101 @@ description: |-
-
-The second phase is ranking, where the scheduler scores feasible nodes to find the best fit.
-Scoring is primarily based on bin packing, which is used to optimize the resource utilization
-and density of applications, but is also augmented by affinity and anti-affinity rules.
-Once the scheduler has ranked enough nodes, the highest ranking node is selected and
-added to the allocation plan.
-
-When planning is complete, the scheduler submits the plan to the leader which adds
-the plan to the plan queue. The plan queue manages pending plans, provides priority
-ordering, and allows Nomad to handle concurrency races. Multiple schedulers are running
-in parallel without locking or reservations, making Nomad optimistically concurrent.
-As a result, schedulers might overlap work on the same node and cause resource
-over-subscription. The plan queue allows the leader node to protect against this and
-do partial or complete rejections of a plan.
+[![Nomad Evaluation Flow][img-eval-flow]][img-eval-flow]
+
+The lifecycle of an evaluation begins with an event causing the evaluation to
+be created. Evaluations are created in the `pending` state and are enqueued
+into the evaluation broker. There is a single evaluation broker which runs on
+the leader server. The evaluation broker is used to manage the queue of pending
+evaluations, provide priority ordering, and ensure at least once delivery.
+
+Nomad servers run scheduling workers, defaulting to one per CPU core, which are
+used to process evaluations. The workers dequeue evaluations from the broker,
+and then invoke the appropriate scheduler as specified by the job. Nomad ships
+with a `service` scheduler that optimizes for long-lived services, a `batch`
+scheduler that is used for fast placement of batch jobs, a `system` scheduler
+that is used to run jobs on every node, and a `core` scheduler which is used
+for internal maintenance. Nomad can be extended to support custom schedulers as
+well.
+
+Schedulers are responsible for processing an evaluation and generating an
+allocation _plan_. The plan is the set of allocations to evict, update, or
+create. The specific logic used to generate a plan may vary by scheduler, but
+generally the scheduler needs to first reconcile the desired state with the
+real state to determine what must be done. New allocations need to be placed
+and existing allocations may need to be updated, migrated, or stopped.
+
+Placing allocations is split into two distinct phases, feasibility checking and
+ranking. In the first phase the scheduler finds nodes that are feasible by
+filtering unhealthy nodes, those missing necessary drivers, and those failing
+the specified constraints.
+
+The second phase is ranking, where the scheduler scores feasible nodes to find
+the best fit. Scoring is primarily based on bin packing, which is used to
+optimize the resource utilization and density of applications, but is also
+augmented by affinity and anti-affinity rules. One anti-affinity rule
```

Comment:
"bin packing" used in previous prose vs "bin-packing" in your proposed replacement.  Which is preferred?